pkg/reconciler/ingress/resources/virtual_service.go

```diff
@@ -84,6 +84,12 @@ func MakeIngressVirtualService(ia v1alpha1.IngressAccessor, gateways map[v1alpha
 
 // MakeMeshVirtualService creates a mesh Virtual Service
 func MakeMeshVirtualService(ia v1alpha1.IngressAccessor) *v1alpha3.VirtualService {
+	hosts := getHosts(ia)
+	if ia.GetSpec().Visibility == v1alpha1.IngressVisibilityClusterLocal {
```

Comment:
I think this isn't correct, since we should only expose cluster-local rules into `-mesh` VirtualService.

Can we please update this to always expand the hosts?  Thanks

Reply:
For non cluster-local service, {virtual service}-mesh is not necessary as traffic is handled by {virtual service}. So I didn't expand the hosts for both in my changes.
For example, httpproxy is accessible externally, and hello world is cluster-local visibility. Following virtual services worked.
```
helloworld-test-image        [knative-serving/cluster-local-gateway knative-serving/knative-ingress-gateway]   [helloworld-test-image.default helloworld-test-image.default.svc helloworld-test-image.default.svc.cluster.local e3fb23448284135ab50e59f2c00e861f.probe.invalid]                                         55m
helloworld-test-image-mesh   [mesh]                                                                            [helloworld-test-image.default helloworld-test-image.default.svc helloworld-test-image.default.svc.cluster.local 141a00decf01e33d31c9f1d1c4d603f7.probe.invalid]                                         55m
httpproxy-test-image         [knative-serving/cluster-local-gateway knative-serving/knative-ingress-gateway]   [httpproxy-test-image.default httpproxy-test-image.default.example.com httpproxy-test-image.default.svc httpproxy-test-image.default.svc.cluster.local 05bdecf2cb547fd7a705bcc292481890.probe.invalid]   55m
httpproxy-test-image-mesh    [mesh]                                                                            [httpproxy-test-image.default.svc.cluster.local 3af9f053d4085250351de5b6efc16ba6.probe.invalid]                                                                                                          55m
``` 
I'm not sure whether I misunderstood your comments.
BTW, I couldn't log in knative workspace of slack because my phone didn't receive verification code. Could u please help to disable 2FA for my account (yu2003w@hotmail.com) if you're owner of the workspace?

Reply:
@yu2003w a public Service also should be exposed locally:  beside being public it will otherwise behave the same as a cluster-local Service.  What that means is that the ServiceToService tests should work the same even if `helloworld` is public.  Which is why for `-mesh` the VirtualService should be the same in both cases.

Reply:
Yes, I missed the cases of ServiceToServiceCall when both services were public.
It seems ServiceToServiceCall in e2e tests are public service call to cluster-local service.
Do we need to add cases to test both public services?